### PR TITLE
fix: teardown fixture for tests and update pre-commit-config

### PR DESCRIPTION
**Pull request by Coniferish**
Files were being created as a side effect from running tests in `test_unstructured/metrics/test_evaluate.py`. The updated decorator removes the created directory and its files after the tests run.

Testing
on the main branch, run `make test` or `pytest test_unstructured/metrics/test_evaluate.py` and files will be created. On this branch no files are created

### CODE REVIEW

File: test_unstructured/metrics/test_evaluate.py [hunk 1]
```diff
@@ -28,12 +29,14 @@ def _cleanup_after_test():
...
+
+    for directory in os.listdir(TESTING_FILE_DIR):
+        if pattern.match(directory):
+            export_dir = os.path.join(TESTING_FILE_DIR, directory)
 
-    # Cleanup the directory and file
-    if os.path.exists(export_dir):
-        shutil.rmtree(export_dir)
+            # Cleanup the directory and its contents
+            shutil.rmtree(export_dir)
```
[the remark this answers is not in the document]
Comment (Coniferish) [reply test_unstructured/metrics/test_evaluate.py:39]: This same issue arose a little while ago with creating a _directory_ named `test_evaluate_results_cct`. Recently @Klaijan added a new test that created a slightly different directory, `test_evaluate_results_cct_txt`. Both contained files in the directories and both directories (and their files) should be removed after the tests run. So, it's intended to look for the matching directories and remove everything.

So... for a test, I'd create a directory and then remove the directory and check that it doesn't exist?

[the remark this answers is not in the document]
Comment (scanny) [reply test_unstructured/metrics/test_evaluate.py:39]: Hmm, okay, I'm starting to understand the situation a little better now.

Well, maybe we don't need a test, but the code needs to be clearer and better documented. If this doesn't work or deletes all the example-docs or something I expect we'll notice it :)

I'd say the following:
1. Extract this part to its own function, can be immediately below or even nested as far as I'm concerned, but `_cleanup_after_test()` is a legitimate fixture that may need to do other things later so its docstring isn't the right place to describe what this passage does. I'd call the extracted function something like `_remove_generated_evaluation_directories()` or whatever it actually is doing. Give that some thought and make it as good as you can based on what you know of those directories/files and their purpose.
2. Give it a nice healthy docstring in which you explain what these files are, how they get there, what they're used for, and that they can be in two different directories and this function nukes them both.
3. Add a comment above the `pattern = ...` line that indicates the pattern matches the directories mentioned in the docstring, even though it appears on first glance to match filenames. Even better, if there are just the two, don't use regex at all and just provide a tuple of the two names and use `if dirname in generated_directories:`.
4. Only get subdirectories if that's all you want, something like this:
```python
subdirs = (d for d in os.scandir(folder_path) if d.is_dir())
for d in subdirs:
    if d.name in target_dir_names:
        shutil.rmtree(d.path)
```

If there is more cleanup that needs to be done later another function can be added for that and called from the fixture without having to interact with this new one.

File: test_unstructured/metrics/test_evaluate.py [hunk 2]
```diff
@@ -56,15 +56,20 @@
...
-    os.path.join(TESTING_FILE_DIR, UNSTRUCTURED_OUTPUT_DIRNAME)
-    export_dir = os.path.join(TESTING_FILE_DIR, "test_evaluate_results_cct")
+    def remove_generated_directories():
+        """Remove directories created from running tests"""
+        # Directories to be removed:
+        target_dir_names = ["test_evaluate_results_cct", "test_evaluate_results_cct_txt"]
+        subdirs = (d for d in os.scandir(TESTING_FILE_DIR) if d.is_dir())
+        for d in subdirs:
+            if d.name in target_dir_names:
+                shutil.rmtree(d.path)
```
Comment (scanny) [inline test_unstructured/metrics/test_evaluate.py:70]: Move this up to the top of the function, line 60 (still needs the blank line before I believe, then keep the yield and the call to this function together. Separating those two by a blank line is expressive I think, but that's the core logic, yield (run the test) and then remove the directories. The definition of the `remove_generated_directories()` function is just defined nested here for convenience and somewhat lower complexity (only used here). The core code should look the same as if that function was defined elsewhere, like at the module level as a peer to this.

**Review by scanny — APPROVED** [review]
Small change, but approved in advance. Let me now when it passes again and I'll merge it :)

**Pull request merged**